File: JavaScript/src/Internals/Functions/DateTimes/Function_MONTH.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';


class Function_MONTH extends Function_1 {
    get Name() {
        return "Month";
    }

    constructor(a) {
        super(a);
    }

    evaluate(engine, tempParameter) {
        let args1 = this.getDate_1(engine, tempParameter);
        if (args1.IsError) { return args1; }
        if (args1.DateValue.Month == null) {
            return this.functionError();
        }
        return Operand.Create(args1.DateValue.Month);
    }
}

export { Function_MONTH };

File: csharp/ToolGood.Algorithm/Internals/Functions/DateTimes/Function_MONTH.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.DateTimes
{
	internal sealed class Function_MONTH : Function_1
    {
        public Function_MONTH(FunctionBase func1) : base(func1)
        {
        }

        public override string Name => "Month";

        public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = GetDate_1(engine, tempParameter);
			if (args1.IsError) { return args1; }
            if (args1.DateValue.Month == null) {
                return FunctionError();
            }
            return Operand.Create((int)args1.DateValue.Month.Value);
        }

    }

}
